support/signal.c

```diff
@@ -317,21 +419,22 @@ setup_pipes (signal_info** signals, int count, struct pollfd *fd_structs, int *c
 
 		h = signals [i];
 
-		if (mph_int_get (&h->pipecnt) == 0) {
+		if (h->pipecnt == 0) { // First pipe for this signal_info
```

Comment:
This shouldn't be changed.

Upon further consideration, here's another two "design guidelines" for this code, that at least appear to partially be present in the existing code ;-)

1. If a member is shared between `default_handler()` and *anything else*, it should *probably* be atomic. This would support your previous change to use `mph_int_set()` to update `h->signum` (above), as `h->signum` is also used in `default_handler()`.
2. If a member is used atomically anywhere, it should be used atomically *everywhere*, so as to stop making my head hurt regarding whether the atomic read is actually necessary. (Extra use of atomics is *fine*, except for additional performance issues, which likely aren't an issue here. Not enough use of atomics can break everything. Consequently, the "stop making my head hurt" answer is "be consistent.")